File: Chapter06/Lesson1/Exercise1/Partial/CS/PieChart/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.Collections;
using System.Drawing.Drawing2D;

namespace PieChart
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Draw(object sender, PaintEventArgs e)
        {
            Draw();
        }

        private void Draw()
        {
            ArrayList data = new ArrayList();
            data.Add(new PieChartElement("East", (float)50.75));
            data.Add(new PieChartElement("West", (float)22));
            data.Add(new PieChartElement("North", (float)72.32));
            data.Add(new PieChartElement("South", (float)12));
            data.Add(new PieChartElement("Central", (float)44));

            chart.Image = drawPieChart(data, new Size(chart.Width, chart.Height));
        }

        private Image drawPieChart(ArrayList elements, Size s)
        {
            //TODO: Complete this method.
        }
    }
}

File: Chapter06/Lesson1/Exercise1/Partial/VB/PieChart/Form1.vb
Imports System.Collections
Imports System.Drawing.Drawing2D

Public Class Form1
    Private Sub Draw(ByVal sender As Object, ByVal e As PaintEventArgs) Handles chart.Paint
        Draw()
    End Sub

    Private Sub Draw()
        Dim data As ArrayList = New ArrayList
        data.Add(New PieChartElement("East", 50.75))
        data.Add(New PieChartElement("West", 22))
        data.Add(New PieChartElement("North", 72.32))
        data.Add(New PieChartElement("South", 12))
        data.Add(New PieChartElement("Central", 44))
        chart.Image = drawPieChart(data, New Size(chart.Width, chart.Height))
    End Sub

    Private Function drawPieChart(ByVal elements As ArrayList, ByVal s As Size) As Image
        ' TODO: Complete this method
    End Function
End Class
